### ISSUE TITLE
Enable Java step definitions to call other steps indirectly by just providing a string.

### ISSUE THREAD

**Issue by epabst** [cucumber-attic/cuke4duke#26]
I don't see any way to pass a String in to cuke4duke to enable calling other steps indirectly.

**Comment by aslakhellesoy** [comment]
In cuke4duke/java, step definitions are plain old java methods. You can just invoke the method directly. If the method is on another step definition class, just declare a dependency on that class in your constructor, and picocontainer will pass it in when the step definition object in instantiated. If you're using spring, just declare dependencies between step definition classes using regular spring DI techniques.

Does this answer your question?

**Comment by epabst** [comment]
The problem is that I have several steps that might need to be delegated to, and I'd like to use the regular expression matching logic to determine which one, parse the remaining string correctly, and call the method.  For example, I have the following possible strings:

a birth of "10 Jan 1959" w/o a place
a marriage-like event of "5 May 2000" in "Paris, France"
a graduation event w/o a date in "Atlanta, GA"
...

I'd like to be able to match "(a (?:birth|._event)) in \"(._?)\"" and have it figure out the place, and delegate to whatever matches group 1.  In Ruby, it's totally straightforward to do that, but in Java, it doesn't seem possible.

**Comment by aslakhellesoy** [comment]
I see. So you want to call-by-string, and have Cucumber pick the right one for you? You need to do that via a method (let's assume we had a Given(String step) and similar When ant Then methods).

This could be on a superclass, or it could be on a special object that your step def class could get a ref to via DI. I like DI better than inheritance.

So what if you could do this:

```
@Given("foo")
public void foo(String theFoo) {
    this.steps.Given("Ping " + theFoo);
    this.steps.Given("Pong " + theFoo);
}
```

Is this what you're after?

**Comment by epabst** [comment]
Yes! That would be perfect. :)

**Comment by aslakhellesoy** [comment]
Enable Java step definitions to call other steps indirectly by just providing a string. Closed by 6a8850173cddba1e8268ec6707a1847fa70715b3

**Comment by aslakhellesoy** [comment]
Wiki updated: http://wiki.github.com/aslakhellesoy/cuke4duke/calling-steps-from-stepdefinitions
